src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -0,0 +1,266 @@
+        }
+
+        public void WriteValue(Guid value)
+        {
+            this.writer.WriteStringValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(decimal value)
+        {
+            if (this.isIeee754Compatible)
+            {
+                this.writer.WriteStringValue(value.ToString(CultureInfo.InvariantCulture));
+            }
+            else
+            {
+                this.writer.WriteNumberValue(value);
+            }
+
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(DateTimeOffset value)
+        {
+            this.writer.WriteStringValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(TimeSpan value)
+        {
+            string stringValue = EdmValueWriter.DurationAsXml(value);
+            this.writer.WriteStringValue(stringValue);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(Date value)
+        {
+            this.writer.WriteStringValue(value.ToString());
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(TimeOfDay value)
+        {
+            this.writer.WriteStringValue(value.ToString());
+        }
+
+        public void WriteValue(byte value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(sbyte value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(string value)
+        {
+            if (value == null)
+            {
+                this.writer.WriteNullValue();
+            }
+            else
+            {
+                this.writer.WriteStringValue(value);
+            }
+
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(byte[] value)
+        {
+            this.writer.WriteBase64StringValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteRawValue(string rawValue)
+        {
+            // TODO: Utf8JsonWriter.WriteRawValue supported from .NET 6
+            // fallback to write to buffer directly as UTF8-encoded bytes
+
+            this.Flush(); // ensure we don't write to the stream while there are still pending data in the buffer
+            this.outputStream.Write(encoding.GetBytes(rawValue));
+            this.FlushIfBufferThresholdReached();
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (this.disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                if (!this.leaveStreamOpen)
+                {
+                    this.outputStream.Dispose();
+                }

```

Comment:
```suggestion
                }

                this.writer.Dispose();
```

I *think* we need to dispose the writer? I don't see anything that says we *shouldn't*, and it's a relatively new class, so I would expect the documentation to be accurate, and for the class to not implement `IDisposable` if it doesn't need to.

Reply:
Yeah, I should dispose it. It should be disposed before the stream is disposed because it will flush the stream contents.

I found the documentation to be a bit misleading. It says that the [Dispose method releases all resources used by the instance](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonwriter.dispose?view=net-6.0#system-text-json-utf8jsonwriter-dispose). But it doesn't really release the stream, it just flushes.

Reply:
Interesting, thanks for the pointer